shell/cloud-credential/vmwarevsphere.vue

```diff
@@ -69,7 +77,8 @@ export default {
           min="1"
           max="65535"
           :mode="mode"
-          @update:value="value.setData('vcenterPort', $event);"
+          :rules="fvGetAndReportPathRules('decodedData.vcenterPort')"
+          @update:value="$emit('valueChanged', 'vcenterPort', $event)"
```

Comment:
There is something wrong with the form validation for this input specifically:

https://github.com/user-attachments/assets/0bb0c5ee-dbd8-44b0-8a76-798ebcacf995

I noticed a couple issues: 
1) using `type="number"` here leads to an empty `value` prop when the user types in non-numbers
2) the `portNumber` validator validates that a number is within the range 1-65535, but it doesnt validate that a value is a number: it should be paired with `requiredInt` or `isInteger` validators

Reply:
We use type="number: for ports in a couple places and I think it's quite awkward. How do you feel about dropping that?
Also, I can't type non-number values into that field. Are you doing anything special to do that?

Reply:
I think it would be fine to drop it, I can't imagine someone setting a port using the arrow buttons. I didn't do anything special to type in it.